src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -189,32 +209,32 @@
   ([database :- i/DatabaseInstance db-metadata]
    ;; determine what's changed between what info we have and what's in the DB
    (let [db-tables               (table-set db-metadata)
-         our-metadata            (our-metadata database)
-         strip-desc              (fn [metadata]
-                                   (set (map #(dissoc % :description) metadata)))
+         name+schema->db-table   (m/index-by (juxt :name :schema) db-tables)
```

Comment:
I took a stab at refactoring this sync step a bit.
Details in the following comments